types/mock-fs/mock-fs-tests.ts

```diff
@@ -1,85 +1,156 @@
+import mock = require('mock-fs');
```

Comment:
Whey most of the mock tests were removed, are they no longer valid?

Reply:
Looking at this again (it's been some time since I've written this), I combined parts of the old tests into one larger one (and added some of my own) to check nesting. 

With that said, looking at it again, I think I'll re-add the v3 tests, but I will strip them out of the functions definitions. I have a separate repo that runs this test file, so the functions need to be called when this file is  `require`'d. I've had a few problems in the past with type definitions typing things that don't exist (not for `mock-fs` though), so I created the runnable version to make sure the functions can be called without errors and work like I think they do.

`mock.fs()` no longer exists in v4 (for now), so the test for that was removed.

I've also realized that I've forgotten to test the second parameter on `mock()`, so I'll add that too. Thanks for this mention since I would've missed that otherwise. The type definitions were part of my private repo and was written a bit independently of the official type definitions, which is why certain things ended up so different.